src/main/scala-2.12/scala/collection/Factory.scala

```diff
@@ -0,0 +1,16 @@
+package scala.collection
+
+import scala.collection.generic.CanBuildFrom
+
+trait Factory[-A, +C] extends Any {
+  def newBuilder(): mutable.Builder[A, C]
+  def fromSpecific(it: TraversableOnce[A]): C
+}
+
+object Factory {
+  implicit def fromCanBuildFrom[A, C](implicit cbf: CanBuildFrom[Nothing, A, C]): Factory[A, C] =
+    new Factory[A, C] {
+      def newBuilder(): mutable.Builder[A, C] = cbf()
+      def fromSpecific(it: TraversableOnce[A]): C = (cbf() ++= it).result()
+    }
+}
```

Comment:
This file is already in `src/main/scala-2.11_2.12/scala/collection/Factory.scala`, no?